src/applications/common/schemaform/fields/AutosuggestField.jsx

```diff
@@ -56,19 +60,31 @@ export default class AutosuggestField extends React.Component {
 
   componentDidMount() {
     if (!this.props.formContext.reviewMode) {
-      const getOptions = this.props.uiSchema['ui:options'].getOptions;
-      if (getOptions) {
-        getOptions().then(options => {
-          if (!this.unmounted) {
-            this.setState({ options, suggestions: this.getSuggestions(options, this.state.input) });
-          }
-        });
-      }
+      this.getOptions();
     }
   }
 
   componentWillUnmount() {
     this.unmounted = true;
+    this.debouncedGetOptions.cancel();
+  }
+
+  getOptions = (inputValue) => {
+    const getOptions = this.props.uiSchema['ui:options'].getOptions;
+    if (getOptions) {
+      // If getOptions returns a promise, when it resolves, set the options to the result.
+      // getOptions may also set the options directly. This is useful if it's a debounced function.
+      const returnedPromise = getOptions(inputValue, this.setOptions);
```

Comment:
Right, exactly. I want to say previously, it was set up to only work for returning a promise, but I found that wasn't always going to work. It's been a while, so I forget the exact reasons, I'm afraid.

:thinking: Actually, now that I think about it, this might be a vestige of debouncing in the `formConfig` instead of in here. Maybe we don't need it now?